src/convert/metadataConverter.ts

```diff
@@ -87,9 +89,9 @@ export class MetadataConverter {
             if (destructiveChangesTypes.length) {
               // for each of the destructive changes in the component set, convert and write the correct metadata
               // to each manifest
-              destructiveChangesTypes.map((destructiveChangeType) => {
+              destructiveChangesTypes.map(async (destructiveChangeType) => {
```

Comment:
Why use `.map` instead of `.forEach` here?

Reply:
forEach doesn't work with async!

Reply:
Oh... doi... you're right, I forgot.
How about just a `for ... of...` loop then?

Reply:
you bring up a good point. I honestly thought that `.forEach` and other array builtin methods were faster than a class `for` loop, but as I learned that's actually [backwards and incorrect](https://leanylabs.com/blog/js-forEach-map-reduce-vs-for-for_of/)

Reply:
@WillieRuemmele I'm less concerned about performance and more concerned about readability & maintainability of the code. 

`.map` creates a new array based on the transformations you apply in its callback, if all you want to do is iterate over an array and you're not going to do anything with the new array `.map` creates then better to just use a loop, regardless of performance considerations (we're not doing anything here that requires millisecond improvements). Using methods that aren't intended for what you're trying to do makes it harder for developers to understand what the code _should_ do and leads to bugs like [this one](https://github.com/forcedotcom/source-deploy-retrieve/pull/595#discussion_r838856708).

Reply:
I really intend to return a new array of Promises that gets added to `tasks`.  Having a type on `tasks` would have also prevented the bug.